core-tests/src/test/java/org/mule/runtime/core/internal/logger/CustomLogger.java

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.logger;
+
+import static org.apache.commons.lang3.StringUtils.replaceOnce;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+
+/**
+ * In some tests we were replacing the original logger in a class with a mocked logger via reflection and then setting it back to
+ * the original. This reflective access will not work in Java 17, hence writing a custom logger that will be used with specific
+ * tests.
+ */
+public class CustomLogger implements Logger {
+
+  private final String name;
+  private List<String> messages;
+  private final Logger logger;
+  private boolean customLoggingEnabled;
+
+  public CustomLogger(Logger logger, String name, boolean loggingEnabled) {
+    this.name = name;
+    this.messages = new ArrayList<>();
+    this.logger = logger;
+    this.customLoggingEnabled = loggingEnabled;
+  }
+
+  public void resetLogs() {
+    this.messages = new ArrayList<>();
+  }
+
+  public List<String> getMessages() {
+    return messages;
+  }
+
+  private void addMessage(String message) {
+    messages.add(message);
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public boolean isTraceEnabled() {
+    if (customLoggingEnabled) {
+      return customLoggingEnabled;
+    } else {
+      return logger.isTraceEnabled();
+    }
+  }
+
+  @Override
+  public void trace(String msg) {
+    addMessage(msg);
+    logger.trace(msg);
+  }
+
+  @Override
+  public boolean isDebugEnabled() {
+    if (customLoggingEnabled) {
```

Comment:
Cangt always rely on whats set in original logger for our testss, then we miss some messages and hence tests fail